mongosql/src/mir/optimizer/stage_movement/mod.rs

```diff
@@ -36,9 +36,10 @@ use crate::{
 use std::collections::HashSet;
 
 impl Stage {
```

Comment:
[nit]: Ugh, I hate to do this, but could we call this bounds instead of pagination? This isn't really correct use of the word pagination.

Reply:
Sure. I chose pagination since skip and limit are often used for that.